src/index.js

```diff
@@ -55,11 +56,17 @@ class DropdownTreeSelect extends Component {
     }
   }
 
-  createList = ({ data, simpleSelect, showPartiallySelected, hierarchical }) => {
-    this.treeManager = new TreeManager({ data, simpleSelect, showPartiallySelected, hierarchical })
+  createList = ({ data, simpleSelect, singleSelect, showPartiallySelected, hierarchical }) => {
+    this.treeManager = new TreeManager({ data, simpleSelect, singleSelect, showPartiallySelected, hierarchical })
     return this.treeManager.tree
   }
 
+  initNewProps = ({ data, simpleSelect, singleSelect, showPartiallySelected, hierarchical }) => {
```

Comment:
`createList` was my attempt at DRYing up the code but with `initNewProps`, we can inline that tree creation, something like:

```suggestion
  initNewProps = ({ data, simpleSelect, singleSelect, showPartiallySelected, hierarchical }) => {
    this.treeManager = new TreeManager({ data, simpleSelect, singleSelect, showPartiallySelected, hierarchical })   
    const tags = this.treeManager.getTags()
    this.setState({ tree: this.treeManager.tree, tags })
  }
```